alpha/alpha-core/src/main/java/org/apache/servicecomb/saga/alpha/core/EventScanner.java

```diff
@@ -64,16 +67,32 @@ public void run() {
   private void pollEvents() {
     scheduler.scheduleWithFixedDelay(
         () -> {
+          updateTimeoutStatus();
+          findTimeoutEvents();
+          abortTimeoutEvents();
           saveUncompensatedEventsToCommands();
           compensate();
           updateCompensatedCommands();
           deleteDuplicateSagaEndedEvents();
+          updateTransactionStatus();
         },
         0,
         eventPollingInterval,
         MILLISECONDS);
   }
 
+  private void findTimeoutEvents() {
+    eventRepository.findTimeoutEvents()
+        .forEach(event -> {
+          log.info("Found timeout event {}", event);
+          timeoutRepository.save(txTimeoutOf(event));
```

Comment:
will this create duplicate timeout entries when multiple alphas work on the same event??

Reply:
Yes, it will throw the integrity constraint violation exception when other alphas save the same record in db just like the command behaves.